pysmt/solvers/qelim.py

```diff
@@ -42,3 +48,39 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         the associated resources.
         """
         del self
+
+
+class ShannonQuantifierEliminator(QuantifierEliminator, IdentityDagWalker):
+
+    LOGICS = [pysmt.logics.BOOL]
+
+    def __init__(self, environment, logic=None):
+        IdentityDagWalker.__init__(self, env=environment)
+        self.logic = logic
+
+    def eliminate_quantifiers(self, formula):
+        logic = get_logic(formula, self.env)
+        if not logic <= pysmt.logics.BOOL:
+            raise NotImplementedError("Shannon quantifier elimination only "\
```

Comment:
Isn't this too strong?
Shouldn't we check that the quantified variables are Boolean? The expression:
``` Exists x. x & (r < 5)``` becomes ```(False & (r < 5)) | (True & (r < 5))```, no?
Am I missing something?